### ISSUE TITLE
Disappearing contours bug and crash 

### ISSUE THREAD

**Issue by pathumego** [trufont/trufont#202]
While trying to reproduce #197 This happend. Please look at the video attached.
https://youtu.be/1hRppqojPuk

Posting parts of the output
```
Internal error: illegal line segment point count: 2

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/defconQt-0.3.0-py3.5.egg/defconQt/glyphView.py", line 823, in paintEvent
    self.drawFillAndStroke(painter, glyph, layerName)
  File "/usr/local/lib/python3.5/site-packages/defconQt-0.3.0-py3.5.egg/defconQt/glyphView.py", line 720, in drawFillAndStroke
    partialAliasing=partialAliasing)
  File "/usr/local/lib/python3.5/site-packages/defconQt-0.3.0-py3.5.egg/defconQt/util/drawing.py", line 297, in drawGlyphFillAndStroke
    contourPath = glyph.getRepresentation("defconQt.NoComponentsQPainterPath")
  File "/Users/Pathum/src/github.com/trufont/trufont/src/defcon/Lib/defcon/objects/base.py", line 337, in getRepresentation
    representation = factory["factory"](self, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/defconQt-0.3.0-py3.5.egg/defconQt/representationFactories/glyphViewFactory.py", line 17, in NoComponentsQPainterPathFactory
    glyph.draw(pen)
  File "/Users/Pathum/src/github.com/trufont/trufont/src/defcon/Lib/defcon/objects/glyph.py", line 369, in draw
    self.drawPoints(pointPen)
  File "/Users/Pathum/src/github.com/trufont/trufont/src/defcon/Lib/defcon/objects/glyph.py", line 379, in drawPoints
    contour.drawPoints(pointPen)
  File "/usr/local/lib/python3.5/site-packages/defconQt-0.3.0-py3.5.egg/defconQt/objects/defcon.py", line 238, in drawPoints
    pointPen.endPath()
  File "/usr/local/lib/python3.5/site-packages/ufoLib-1.2-py3.5.egg/ufoLib/pointPen.py", line 132, in endPath
    self._flushContour(segments)
  File "/usr/local/lib/python3.5/site-packages/ufoLib-1.2-py3.5.egg/ufoLib/pointPen.py", line 178, in _flushContour
    assert len(points) == 1, "illegal line segment point count: %d" % len(points)
AssertionError: illegal line segment point count: 2
None
```

```
<TPoint position: (212.19512195121956, 371.95121951219517) type: move smooth: False name: None identifier: None>
<TPoint position: (212.19512195121956, 371.95121951219517) type: None smooth: False name: None identifier: None>
<TPoint position: (-43.902439024390105, 397.56097560975627) type: None smooth: False name: None identifier: None>
<TPoint position: (60.670731707317145, 207.6219512195123) type: curve smooth: True name: None identifier: None>
<TPoint position: (165.2439024390244, 17.682926829268354) type: None smooth: False name: None identifier: None>
<TPoint position: (167.37804878048792, -82.62195121951208) type: None smooth: False name: None identifier: None>
<TPoint position: (641.1585365853659, 378.3536585365855) type: curve smooth: True name: None identifier: None>
<TPoint position: (923.8529158074643, 653.4076491310668) type: None smooth: False name: None identifier: None>
<TPoint position: (295.42682926829275, -5.7926829268292295) type: None smooth: False name: None identifier: None>
<TPoint position: (340.2439024390244, 444.5121951219512) type: curve smooth: False name: None identifier: None>
<TPoint position: (340.2439024390244, 444.5121951219512) type: None smooth: False name: None identifier: None>
<TPoint position: (446.61717672952364, 675.1368652475128) type: None smooth: False name: None identifier: None>
<TPoint position: (212.19512195121956, 817.9878048780488) type: curve smooth: True name: None identifier: None>
<TPoint position: (150.30487804878044, 675.0) type: None smooth: False name: None identifier: None>

<TPoint position: (116.15853658536594, 478.65853658536594) type: line smooth: False name: None identifier: None>
<TPoint position: (210.06097560975627, 369.8170731707319) type: line smooth: False name: None identifier: None>
<TPoint position: (210.06097560975627, 369.8170731707319) type: None smooth: False name: None identifier: None>
<TPoint position: (150.30487804878067, 260.9756097560978) type: None smooth: False name: None identifier: None>
<TPoint position: (210.06097560975627, 369.8170731707319) type: curve smooth: True name: None identifier: None>
<TPoint position: (269.8170731707319, 478.65853658536594) type: None smooth: False name: None identifier: None>
<TPoint position: (178.04878048780483, 544.8170731707319) type: None smooth: False name: None identifier: None>
<TPoint position: (178.04878048780483, 544.8170731707319) type: curve smooth: True name: None identifier: None>
<TPoint position: (120.42682926829275, 568.2926829268295) type: None smooth: False name: None identifier: None>

<TPoint position: (-92.98780487804868, 805.1829268292686) type: move smooth: False name: None identifier: None>

<TPoint position: (103.3536585365855, 500.0) type: move smooth: False name: None identifier: None>
<TPoint position: (103.3536585365855, 500.0) type: None smooth: False name: None identifier: None>
<TPoint position: (-596.6463414634145, 7.012195121951436) type: None smooth: False name: None identifier: None>
<TPoint position: (-144.20731707317066, 139.32926829268308) type: curve smooth: True name: None identifier: None>
<TPoint position: (308.2317073170732, 271.64634146341473) type: None smooth: False name: None identifier: None>
<TPoint position: (438.4146341463413, 186.28048780487802) type: None smooth: False name: None identifier: None>
<TPoint position: (321.0365853658536, 248.17073170731715) type: curve smooth: True name: None identifier: None>
<TPoint position: (203.65853658536594, 310.06097560975627) type: None smooth: False name: None identifier: None>
<TPoint position: (156.70731707317077, 450.91463414634154) type: None smooth: False name: None identifier: None>
<TPoint position: (156.70731707317077, 450.91463414634154) type: curve smooth: False name: None identifier: None>
<TPoint position: (348.780487804878, 555.4878048780488) type: line smooth: False name: None identifier: None>

**********
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/defconQt-0.3.0-py3.5.egg/defconQt/glyphCollectionView.py", line 585, in paintEvent
  File "/Users/Pathum/src/github.com/trufont/trufont/src/defcon/Lib/defcon/objects/base.py", line 337, in getRepresentation
    representation = factory["factory"](self, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/defconQt-0.3.0-py3.5.egg/defconQt/representationFactories/qPainterPathFactory.py", line 7, in QPainterPathFactory
  File "/Users/Pathum/src/github.com/trufont/trufont/src/defcon/Lib/defcon/objects/glyph.py", line 369, in draw
    self.drawPoints(pointPen)
  File "/Users/Pathum/src/github.com/trufont/trufont/src/defcon/Lib/defcon/objects/glyph.py", line 379, in drawPoints
    contour.drawPoints(pointPen)
  File "/usr/local/lib/python3.5/site-packages/defconQt-0.3.0-py3.5.egg/defconQt/objects/defcon.py", line 238, in drawPoints
  File "/usr/local/lib/python3.5/site-packages/ufoLib-1.2-py3.5.egg/ufoLib/pointPen.py", line 132, in endPath
  File "/usr/local/lib/python3.5/site-packages/ufoLib-1.2-py3.5.egg/ufoLib/pointPen.py", line 178, in _flushContour
AssertionError: illegal line segment point count: 2
/Users/Pathum/src/github.com/trufont/trufont.command: line 3: 27568 Abort trap: 6         
```

**Comment by adrientetar** [comment]
Great, thanks for this! Maybe I can reproduce that bug after all with these informations.

**Comment by pathumego** [comment]
I can email you the whole thing, but I think its the same repeating

**Comment by adrientetar** [comment]
If you update trufont and defcon, can you repro still?

**Comment by pathumego** [comment]
Not anymore, but I don't know how to use that output to reproduce error, Im just trying to do the same thing in UI.

**Comment by adrientetar** [comment]
What do you mean? I pushed a few fixes for the issues you showed in the video but I couldn't get the `AssertionError: illegal line segment point count: 2` traceback where there is a line segment at start with a trailing offCurve. I haven't reproduced that so far.

**Comment by adrientetar** [comment]
This issue has different symptoms but same root cause as #195, #169, #160, #35 (they are different manifestations of the similar erratic segment). All of these happen because there is a line start point and an offCurve at the end of contour, which makes a 2-points line segment (illegal).  
Normally under trailing offCurve the start point should always be a move (open contour), so I'd be interested in a way to reproduce this `AssertionError: illegal line segment point count: 2` crash reliably. When I did what you do in the video I got a crash but with a different error.
